### PR TITLE
Newsletters: add 179 (2021-12-15)

### PR DESCRIPTION
**Pull request by harding**
- [x] (Monday) `Bitcoin Core #20295` rpc: getblockfrompeer @Xekyo 
- [x] (Monday) `BOLTs #906` introduce feature bit to gate new channel_type feature @dongcarl 
- [x] (Monday) Monthly section or sections @bitschmidty 
- [x] (Tuesday) Update lede and releases/RCs, plus add topic links @harding

### CODE REVIEW

File: _posts/en/newsletters/2021-12-15-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+permalink: /en/newsletters/2021/12/15/
+name: 2021-12-15-newsletter
+slug: 2021-12-15-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a proposal to allow relay of
+transactions with zero-value outputs in some cases and summarizes a
+discussion about preparing LN for the adoption of PTLCs.  FIXME:harding
```
Comment (lightning-developer) [inline _posts/en/newsletters/2021-12-15-newsletter.md:12]: This seems very high level. As far as I understood the posts on the mailing list the discussion was mainly about preparing the communication protocol for setting up and settling PTLCs with as little roundtrips as possible. Maybe we could write something like: 

> ... summarizes a discussion about preparing LN for a stepwise upgrade to PTLCs.

File: _posts/en/newsletters/2021-12-15-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  transaction relay policy that nodes use to discourage users from
+  creating UTXOs that don't make economic sense to spend.  UTXOs need to
+  be stored by at least some full nodes until spent, and in some cases
+  be able to be retrieved quickly, so allowing *uneconomical outputs*
+  may create problems for no good reason.
+
+    However, there may be a use for zero-value outputs in [CPFP][topic
+    cpfp] fee bumping where the none of the funds from the transaction
+    being fee bumped can be spent---all the funds used for fee bumping
+    need to come from separate UTXOs, such as in [eltoo][topic eltoo].
```
Comment (lightning-developer) [inline _posts/en/newsletters/2021-12-15-newsletter.md:31]: aren't anchor outputs the more relevant example here? Anchor outputs are already in practice while eltoo is far from being implemented.

File: _posts/en/newsletters/2021-12-15-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+- [Bitcoin Core #23486][] updates the `decodescript` RPC to only return
+  the P2SH or P2WSH addresses for a script if that script can be used,
+  respectively, with P2SH or P2WSH.
+
+- [BOLTs #940][] deprecates the announcement and parsing of Tor
+  v2 onions in `node_announcements`. [Rust-Lightning #1204][], also
+  merged this week, updates that implementation to follow this updated
+  specification.
+
```
Comment (lightning-developer) [inline _posts/en/newsletters/2021-12-15-newsletter.md:99]: I added https://github.com/lightning/bolts/pull/918 in a PR in @harding 's repo at https://github.com/harding/bitcoinops.github.io/pull/23 The PR is about dropping the requirement to ratelimit ping and pong messages which ensures higher quality of service for LSPs

**Review by lightning-developer — CHANGES_REQUESTED** [review]
As always a very interesting newsletter I suggested another BOLTs PR and suggest to focus on anchor outputs instead of eltoo in the discussion about fee bumping.

File: _posts/en/newsletters/2021-12-15-newsletter.md (hunk 2 above)
Comment (harding) [reply _posts/en/newsletters/2021-12-15-newsletter.md:31]: I don't think there's any advantage to anchor outputs being zero-value.  The transaction containing them *needs* to be fee-bumped, so they might as well contain at least the dust amount of funds to contribute towards that fee bumping.

By comparison, most transactions in an eltoo or statechain system will remain offchain (even if they can all hypothetically be put onchain) and so they can't each pay some value towards an output without quickly draining the system of value.  This is usually solved by them using some version of `SIGHASH_SINGLE` to allow adding additional outputs, but Jeremy Rubin made a case that avoiding SINGLE and using CPFP instead might avoid RBF pinning attacks.

File: _posts/en/newsletters/2021-12-15-newsletter.md (hunk 3 above)
Comment (harding) [reply _posts/en/newsletters/2021-12-15-newsletter.md:99]: I did think this was minor, but I like the description of how it benefits LSPs.  Merged.  Thanks!

File: _posts/en/newsletters/2021-12-15-newsletter.md (hunk 1 above)
Comment (harding) [reply _posts/en/newsletters/2021-12-15-newsletter.md:12]: I think this point is covered in the extended description.  I prefer to keep the lede as short as possible.

**Review by adamjonas — COMMENTED** [review]
A couple of notes but looking good!

File: _posts/en/newsletters/2021-12-15-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  it to be enabled separately.
+
+- [Bitcoin Core #20295][] adds a new RPC `getblockfrompeer` which permits
+  requesting a specific block from a specific peer manually. The intended use
+  is acquisition of stale chaintips for fork monitoring and research purposes.
+
+- [Bitcoin Core #14707][] updates several RPCs to also include bitcoins
+  received in miner coinbase transactions.  A new
+  `include_immature_coinbase` option to the RPCs allows toggling
+  the inclusion of coinbase transactions before they are mature (have
```
Comment (Xekyo) [inline _posts/en/newsletters/2021-12-15-newsletter.md:149]: Perhaps:

```suggestion
  the inclusion of coinbase transaction outputs before they are mature (have
```

**Review by Xekyo — COMMENTED** [review]
Looking very nice. Got a couple suggestions.

File: _posts/en/newsletters/2021-12-15-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  creating UTXOs that don't make economic sense to spend.  UTXOs need to
+  be stored by at least some full nodes until spent, and in some cases
+  be able to be retrieved quickly, so allowing *uneconomical outputs*
+  may create problems for no good reason.
+
+    However, there may be a use for zero-value outputs in [CPFP][topic
+    cpfp] fee bumping where the none of the funds from the transaction
+    being fee bumped can be spent---all the funds used for fee bumping
+    need to come from separate UTXOs, such as in [eltoo][topic eltoo].
+    Rubin Somsen also [provided][somsen unec] an example of how zero-fee
```
Comment (glozow) [inline _posts/en/newsletters/2021-12-15-newsletter.md:32]: Name misspelled:
```suggestion
    Ruben Somsen also [provided][somsen unec] an example of how zero-fee
```

**Review by glozow — COMMENTED** [review]
Light review, just one comment

File: _posts/en/newsletters/2021-12-15-newsletter.md (hunk 5 above)
Comment (glozow) [reply _posts/en/newsletters/2021-12-15-newsletter.md:32]: source: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-December/019637.html

Comment (harding) [reply _posts/en/newsletters/2021-12-15-newsletter.md:32]: Oof, there are few things that embarrass me more than getting names wrong, thank you for catching before publication!

**Comment by harding** [comment]
Made suggested edits, thanks @lightning-developer, @adamjonas, @Xekyo, and @glozow!  Also made final additions.

**Comment by bitschmidty** [comment]
Small fixup upon final review. Squashed.

**Comment by bitschmidty** [comment]
Final regular newsletter of the year! Thank you @harding @lightning-developer (two weeks in a row!), @Xekyo, @glozow, @adamjonas

**Pull request merged**